### PR TITLE
p2p: Per channel metrics (#3666)

### PR DESCRIPTION
**Pull request by brapse**
Reactors  run in parallel exchanging messages with peers across channels. To understand the behaviour of the system as a whole, it would be helpful that have visibility into the per channel (and thus per reactor) message exchange. With per channel metrics, we can observe when a given reactor has stopped sending/receiving and correlate cpu usage with potential thread saturation.

Cf. #3666

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -45,6 +45,8 @@
     use `make build_c` / `make install_c` (full instructions can be found at
     https://tendermint.com/docs/introduction/install.html#compile-with-cleveldb-support)
   * Use `boltdb` tag to compile Tendermint with bolt db
+- [p2p] [\#3666](https://github.com/tendermint/tendermint/issues/3666)
```
Comment (Liamsi) [inline CHANGELOG_PENDING.md]: No need to change anything but just FYI: you do not need to add the link yourself here. There is a little python script that generates the link to the issue from `[\#3666]`.

**Comment by brapse** [comment]
> Awesome thanks - should this be considered a breaking change? Ie. will anyones current setup break with this change, or does it just add new/more info ?

This should be a non breaking changing. The only concern which might be worth noting is this will create additional metrics and increased data for prometheus server pulling the data in. So long as the peer set is bounded and has low churn, This shouldn't be too big a deal.

**Comment by melekes** [comment]
> So long as the peer set is bounded and has low churn, This shouldn't be too big a deal.

not the case for seed nodes

**Comment by brapse** [comment]
Would it be something in the region of:
`1000s of peers, 10s of channels`?
Which I think is ok to pull and could be queried efficiently with a [rule](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/)

**Pull request merged**